pkg/ebpf/tracer.go

```diff
@@ -104,15 +108,51 @@ func NewFlowFetcher(cfg *FlowFetcherConfig) (*FlowFetcher, error) {
 	}); err != nil {
 		return nil, fmt.Errorf("rewriting BPF constants definition: %w", err)
 	}
-
-	if err := spec.LoadAndAssign(&objects, nil); err != nil {
-		var ve *ebpf.VerifierError
-		if errors.As(err, &ve) {
-			// Using %+v will print the whole verifier error, not just the last
-			// few lines.
-			log.Infof("Verifier error: %+v", ve)
+	// For older kernel (< 5.17) kfree_sbk drop hook doesn't exists
+	if cfg.SupportOldKernel {
+		// Here we define another structure similar to the bpf2go created one but w/o the hooks that does not exist in older kernel
+		// Note: if new hooks are added in the future we need to update the following structures manually
+		type NewBpfPrograms struct {
```

Comment:
This is a neat approach!! can be extended to various hooks and configurations like RTT also maybe!

Reply:
atm there is only skb_free hook that isn't there in older kernel as we run into more cases we can generalize ?

Reply:
yeah if this is only one hook.